go/store/types/value_store.go

```diff
@@ -99,12 +99,19 @@ const (
 )
 
 // newTestValueStore creates a simple struct that satisfies ValueReadWriter
-// and is backed by a chunks.TestStore.
+// and is backed by a chunks.TestStore. Used for testing noms.
 func newTestValueStore() *ValueStore {
 	ts := &chunks.TestStorage{}
 	return NewValueStore(ts.NewView())
 }
 
+// NewInternalValueStore creates a simple struct that satisfies ValueReadWriter
+// and is backed by a chunks.TestStore. Used for dolt operations outside of noms.
+func NewInternalValueStore() *ValueStore {
```

Comment:
Kind of a bad name. ReadOnlyValueStore? The fact that it's internal doesn't really tell me anything useful about its limitations.

Reply:
It's not read only. Maybe `NewMemoryValueStore` would be better. It's just a value store that in memory and should be used if something needs to write something to get a hash (such as a `Ref` used by `ChunkedString`) but we don't want to actually write to disk.

Reply:
Should this just panic if we try to write a value to it? It's hard for me to reason about times when we want a Blob/ChunkedSting instance floating around that has refs in it to a transient `ValueStore` but we know for certain that thing is on the right side of a firewall such that it never gets written to a persisted `ValueStore` or something...

Reply:
So this is really only being used for places where we are displaying data, such as diffs and whatnot. If, in the process of diffing one of the blob types pops up and needs a VRW, we have this one that it can interact with. Everywhere that's on a write path (SQL, merge, import, etc.) is making use of the contextual VRW.